lib/no_brainer/document/types/array.rb

```diff
@@ -0,0 +1,89 @@
+require 'no_brainer/document'
+require 'active_support/core_ext/array/wrap'
+
+module NoBrainer
+  class Array
+    # delegate casting methods to each array element, if defined
+    %i(nobrainer_cast_user_to_model nobrainer_cast_model_to_db nobrainer_cast_db_to_model).each do |method|
+      redefine_singleton_method(method) do |values|
+        ::Array.wrap(values).map do |value|
+          if value.class.respond_to?(method)
+            value.class.__send__(method, value)
+          else
+            value
+          end
+        end
+      end
+    end
+
+    # convenience method to create a TypedArray
+    def self.of(object_type = nil, &object_type_proc)
```

Comment:
This function name is too generic for adding it to built-in classes

Reply:
This is just on `NoBrainer::Array`, it doesn't get added to `::Array`.  It was harder this way, but I didn't want to pollute the core classes.